internal/fileutils/fileutils.go

```diff
@@ -103,3 +105,28 @@ func Lrename(oldPath, newPath string) error {
 
 	return os.Rename(oldPath, newPath)
 }
+
+// LockDir creates a lock file in the specified directory and acquires an exclusive lock on it.
+// It blocks until the lock is available and returns an unlock function to release the lock.
+func LockDir(dir string) (func() error, error) {
+	lockPath := dir + "/.lock"
+	f, err := os.OpenFile(lockPath, os.O_CREATE|os.O_RDWR, 0600)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := unix.Flock(int(f.Fd()), unix.LOCK_EX); err != nil {
+		_ = f.Close()
+		return nil, err
+	}
+
+	unlock := func() error {
+		if err := unix.Flock(int(f.Fd()), unix.LOCK_UN); err != nil {
+			_ = f.Close()
```

Comment:
Warn maybe?

Reply:
I tried to avoid pulling in a dependency on our log package, because this package could be easily used in another repository (if we move it out of `internal/`). We're already returning an error because `Flock` failed, do you expect any benefit from also printing a warning if `Close` failed?